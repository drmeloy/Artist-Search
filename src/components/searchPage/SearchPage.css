@import url('https://fonts.googleapis.com/css?family=Raleway:300&display=swap');
@import url('https://fonts.googleapis.com/css?family=Raleway:800&display=swap');
@import url('https://fonts.googleapis.com/css?family=Montserrat:300&display=swap');

.Form {
  padding-left: 30px;
  display: flex;
  flex-direction: column;

  input {
    font-size: 1.5rem;
  }

  button {
    width: 300px;
    height: 30px;
    margin-left: 30px;
    border-radius: 3px;
  }
}

/* H1 and Headers get Raleway Heavy, font-weight: 800 */
.Header {
  font-family: 'Raleway', sans-serif;
  font-weight: 800;
  font-size: 3rem;
  color: white;
}


/* Sub-Headings get Raleway, font-weight 300 */
.subHeader {
  font-family: 'Raleway', sans-serif;
  font-weight: 300;
  font-size: 3rem;
  color: white;
}


/* Any body text gets Montserrat, font-weight 300 */
.bodyText {
  font-family: 'Montserrat', sans-serif;
  font-weight: 300;
}
