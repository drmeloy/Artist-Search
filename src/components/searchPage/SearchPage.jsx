import React from 'react';
import Artists from '../artists/Artists.jsx';
import { useGetArtists } from '../hooks/artists';
import styles from './SearchPage.css';

const SearchPage = () => {
  const { artists, handleSubmit, query, count, setQuery } = useGetArtists();

  return (
    <>
      <form className={styles.Form} onSubmit={handleSubmit}>
        <p className={styles.subHeader} >SEARCH FOR ARTISTS</p>
        <div>
          <input type='text' value={query} onChange={({ target }) => setQuery(target.value)}></input>
          <button>Search</button>
        </div>
      </form>
      {count && <p className={styles.Count}>{count} results</p>}
      <Artists artists={artists} />
    </>
  );
};

export default SearchPage;
