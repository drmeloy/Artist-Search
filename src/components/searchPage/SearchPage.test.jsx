import React from 'react';
import { shallow } from 'enzyme';
import SearchPage from './SearchPage';

describe('SearchPage component', () => {
  it('matches its snapshot', () => {
    const wrapper = shallow(<SearchPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
