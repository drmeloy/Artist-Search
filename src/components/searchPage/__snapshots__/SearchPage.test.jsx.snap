// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage component matches its snapshot 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p>
      Search for Artists
    </p>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button>
      Search
    </button>
  </form>
  <Artists
    artists={Array []}
  />
</Fragment>
`;
